src/Models/Artificial Intelligence/KongAI.js

```diff
@@ -48,23 +133,50 @@ export default class KongAI extends Player{
             currentQueue = nextQueue;
             depth++;
         }
-        console.log(boardStates,"finished");
+        // for(let state of boardStates["children"]){
+        //     this.identifyDifference(boardStates.PlayerTwo.ActivePawns,state["board"].PlayerTwo.ActivePawns);
+        // }
+        return boardStates;
     }
     getBoardStatePawnMove(pawn,board){
         let newBoardPieceMoves=[]
         for(let pos of board.getAvailableMoves(pawn)){
-            let stateSpace = {};
+
+            let stateSpaceOrigin = {};
             let movePieceBoard = clonedeep(board);
             let pawnClone = clonedeep(pawn);
             movePieceBoard.selectedPawn = pawnClone;
             movePieceBoard.selectedTile = movePieceBoard.grid[pawnClone.Position.x][pawnClone.Position.y];
             movePieceBoard.movePawn(movePieceBoard.grid[pos.x][pos.y]);
-            stateSpace["board"] = movePieceBoard;
-
-            newBoardPieceMoves.push(stateSpace);
+            movePieceBoard.unhighlightTiles();
+            // this.identifyDifference(this.boardStatus.PlayerTwo.ActivePawns,movePieceBoard["board"].PlayerTwo.ActivePawns);
+            let targets = movePieceBoard.selectedPawn.getTargets(movePieceBoard.grid);
+
+            // Account for Attack Moves
+            for(let enemy of targets){
+                let stateSpace = {};
+                let attackPieceBoard = clonedeep(movePieceBoard);
+                attackPieceBoard.showPawnAttack(pawnClone.Position.x,pawnClone.Position.y);
```

Comment:
What is this used for?

Reply:
To handle Attack Sequences if there are any targets in the selected pawn's line of sight